vars/deploy.groovy

```diff
@@ -77,16 +81,41 @@ def tagImageToDeployEnv(ns, userNamespace, imageStreams, tag) {
   }
 }
 
+/* It pause all triggers happening upon config change in DC.
+* Image tagging, changing DC config does not results into
+* immediate update.
+* */
+def pauseDeployments(ns, dcs) {
+  dcs.each { dc ->
+    if(shWithOutput(this, "oc get dc/${dc.metadata.name} -n $ns --ignore-not-found"))
```

Comment:
how would this work in case of multiple pipelines running in parallel where 
1. is doing the actual deployment
2. while other pauses the deployment?

Reply:
for better readabilty could you please move the `shWithOutput` outside if? e.g. 

```groovy
hasDC = shWithOutput(this, "...")
if (hasDC) {
  oc rollout pause .. 
}
```

Reply:
> how would this work in case of multiple pipelines running in parallel where

Not sure, is there any possibility of multiple pipelines updating the same deployment?

In worst case even if it happens user might see mutiple deployments tiggerns, dispite that it would retain the consistency of deployment. right?